privacyidea/lib/eventhandler/requestmangler.py

```diff
@@ -0,0 +1,163 @@
+                        },
+                   ACTION_TYPE.ADD:
+                       {"parameter":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The parameter that should be added or modified.")
+                             },
+                        "value":
+                            {"type": "str",
+                             "required": True,
+                             "description": _("The new value of the parameter. Can contain tags like {0}, {1} for  "
+                                              "the matched sub strings.")
+                            },
+                        "match_parameter":
+                            {"type": "str",
+                             "description": _("The parameter, that should match some values.")
+                            },
+                        "match_pattern":
+                            {"type": "str",
+                             "description": _("The value of the match_parameter. It can contain a regular "
+                                              "expression and '()' to transfer values to the new parameter.")
+                            }
+                        }
+                   }
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action:
+        :param options: Contains the flask parameters g, request, response
+            and the handler_def configuration
+        :type options: dict
+        :return:
+        """
+        ret = True
+        request = options.get("request")
+        handler_def = options.get("handler_def")
+        handler_options = handler_def.get("options", {})
+        parameter = handler_options.get("parameter")
+
+        if parameter:
+            if action.lower() == ACTION_TYPE.DELETE:
+                if parameter in request.all_data:
+                    del(request.all_data[parameter])
+            elif action.lower() == ACTION_TYPE.ADD:
+                value = handler_options.get("value")
+                match_parameter = handler_options.get("match_parameter")
+                match_pattern = handler_options.get("match_pattern")
+                if value:
+                    # We only take action, if we have a value
+                    if not match_parameter:
+                        # simple setting a parameter
+                        request.all_data[parameter] = value
+                    elif match_pattern:
+                        # setting a parameter depending on another value
+                        m = re.match(match_pattern, request.all_data.get(match_parameter))
+                        if m:
+                            # Now we set the new value with the matching tuple
+                            request.all_data[parameter] = value.format(*m.groups())
```

Comment:
In general, one should be careful with user-supplied format strings: http://lucumr.pocoo.org/2016/12/29/careful-with-str-format/
because passing Python objects may make it possible to leak secrets. But in this case, I think we are fine because we only pass strings. Maybe we should add a comment about that?

Reply:
I am not quite sure, how this would be relevant to us. 
In the new parameter we can only add what was already in the old parameter.
I.e. we can only format-in what was already passed in the old parameter.

So you want me to add a comment at this code part, that using a user defined format string in this case is fine?

Reply:
The problem with a user-supplied format string in combination with object arguments is that a rogue admin could try to extract e.g. the ``SECRET_KEY`` with a carefully crafted format string.

So yes, adding a comment would be nice I think, like:
```python
# Using a user-supplied format string can be dangerous: http://lucumr.pocoo.org/2016/12/29/careful-with-str-format/
# but in this case it is fine because no objects are involved, as re.groups() always returns a tuple of strings
```
so if someone decides to extend this in the future, there is a warning :)

Reply:
Done in b534ba2e

Reply:
thanks!